### PR TITLE
Enable named aggregation syntax

### LINKED ISSUE #5294: Feature request: support pandas NamedAgg
pandas 0.25.0 [introduced](https://pandas.pydata.org/pandas-docs/stable/whatsnew/v0.25.0.html#groupby-aggregation-with-relabeling) a new "named aggregation" feature. It would be lovely if dask.dataframe supported `NamedAgg` as well. 

Here's an example, copied from the pandas docs, that works in pandas but not dask.
```python
import numpy as np
import dask.dataframe as dd
import pandas as pd
animals = pd.DataFrame({'kind': ['cat', 'dog', 'cat', 'dog'],
                            'height': [9.1, 6.0, 9.5, 34.0],
                            'weight': [7.9, 7.5, 9.9, 198.0]})
animals = dd.from_pandas(animals, npartitions=1)
animals.groupby("kind").agg(
        min_height=pd.NamedAgg(column='height', aggfunc='min'),
        max_height=pd.NamedAgg(column='height', aggfunc='max'),
        average_weight=pd.NamedAgg(column='weight', aggfunc=np.mean),
   )
#> Traceback (most recent call last):
#> TypeError: agg() got an unexpected keyword argument 'min_height'
```

(This `NamedAgg` feature is absolutely not essential; pandas worked well for years without it.)

### PR DESCRIPTION
**Pull request by ChrisJar**
Closes #5294

### CODE REVIEW

**Comment by GPUtester** [comment]
Can one of the admins verify this patch?

Admins can comment `ok to test` to allow this one PR to run or `add to allowlist` to allow all future PRs from the same author to run.

**Comment by charlesbluca** [comment]
add to allowlist

**Comment by ncclementi** [comment]
@ChrisJar Thank you for the contribution. Would you mind fixing the conflicts so we can get a review in progress?

File: dask/dataframe/groupby.py [hunk 1]
```diff
@@ -1881,7 +1888,9 @@ def get_group(self, key):
         )
 
     @_aggregate_docstring()
```
Comment (rjzamora) [inline dask/dataframe/groupby.py:1890]: This is looking good @ChrisJar - Thanks!

Can we update [_aggregate_docstring](https://github.com/dask/dask/blob/0532bb39094e8ab42525f65926f5aab773f51fbc/dask/dataframe/groupby.py#L1154) to add a description for `**kwargs` (and to clarify that these additional key-word arguments will be interpreted as named aggregations)?

**Review by rjzamora — APPROVED** [review]
This looks good @ChrisJar - Thanks!

Would you be able to run [`dask_cudf` tests](https://github.com/rapidsai/cudf/blob/branch-22.12/python/dask_cudf/dask_cudf/tests/test_groupby.py) locally to confirm that this doesn't cause any unexpected breakages?

**Comment by ChrisJar** [comment]
@rjzamora Thanks for the reviews! I can confirm all dask_cudf tests pass locally

**Pull request merged**